spark/v3.4/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CreateOrReplaceBranchExec.scala

```diff
@@ -41,6 +41,9 @@ case class CreateOrReplaceBranchExec(
   override protected def run(): Seq[InternalRow] = {
     catalog.loadTable(ident) match {
       case iceberg: SparkTable =>
+        if (iceberg.table.currentSnapshot() == null) {
+          throw new UnsupportedOperationException(s"The Iceberg table: $iceberg" + " has no snapshots")
+        }
```

Comment:
I think `iceberg.table.currentSnapshot().snapshotId` below always assume the currentSnapshot is not null but it's not the case. Probably need a way to probably handle that instead of branching out and throw a new exception.  FYI @amogh-jahagirdar

Reply:
Not very familiar with design of iceberg snapshots.  I found many annoying code snippet `snapshot == null` , maybe the issue will be gone if every new table has a default snapshot.  Just my immature thought. :)